### PR TITLE
STUW-140 increased frontent testcoverage

### PR DESCRIPTION
**Pull request by Toubi01**
Dieser Pull request soll die Story [STUW-140](https://stuwork.atlassian.net/browse/STUW-140) erfüllen.

### CODE REVIEW

File: frontend/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
   let fixture;
+  const authService = createServiceSpy('Auth');
+  const boolSubject = new BehaviorSubject<boolean>(false);
+  authService.isAuthenticated$ = boolSubject;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       declarations: [AppComponent],
-      providers: [{ provide: AuthService, useValue: createServiceSpy('Auth') }],
+
```
Comment (Marlaqk) [inline frontend/src/app/app.component.spec.ts:19]: remove empty line

File: frontend/src/app/core/auth.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const stringSubject = new BehaviorSubject<string>('login');
+  oAuthMock.events = stringSubject;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [TestingModule.forRoot()],
+      providers: [{ provide: OAuthService, useValue: oAuthMock }],
+    });
+    service = TestBed.inject(AuthService);
+    router = TestBed.inject(Router);
```
Comment (Marlaqk) [inline frontend/src/app/core/auth.service.spec.ts:21]: Reset all the spies before every test run. If you don't do it you could have tests leaking into each other.

````
    oAuthMock.loadDiscoveryDocument.calls.reset();
    oAuthMock.tryLogin.calls.reset();
    oAuthMock.hasValidAccessToken.calls.reset();
    oAuthMock.silentRefresh.calls.reset();
````

File: frontend/src/app/core/auth.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should runInitialLoginSequence with getValidAccessToken', async () => {
+    oAuthMock.loadDiscoveryDocument.and.returnValue(Promise.resolve());
+    oAuthMock.tryLogin.and.returnValue(Promise.resolve());
+    oAuthMock.hasValidAccessToken.and.returnValue(true);
+    service.isDoneLoading$.subscribe((value) => expect(value).toBeTruthy());
+    await service.runInitialLoginSequence();
```
Comment (Marlaqk) [inline frontend/src/app/core/auth.service.spec.ts:33]: For better readability you could add an empty line between the spies and the service call.

File: frontend/src/app/notes/notes.component.spec.ts [hunk 4]
```diff
@@ -13,7 +12,7 @@ describe('NotesComponent', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
       declarations: [NotesComponent],
-      providers: [{ provide: IdentityService, useValue: identitySpy }],
+      providers: [{ useValue: identitySpy }],
```
Comment (Marlaqk) [inline frontend/src/app/notes/notes.component.spec.ts:15]: remove `identitySpy`. The spy isn't needed anymore.

**Review by Marlaqk — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.component.spec.ts (hunk 1 above)
Comment (Toubi01) [reply frontend/src/app/app.component.spec.ts]: done

File: frontend/src/app/core/auth.service.spec.ts (hunk 2 above)
Comment (Toubi01) [reply frontend/src/app/core/auth.service.spec.ts:21]: added them plus also the `oAuthMock.logOut.calls.reset();` and `oAuthMock.initLoginFlow.calls.reset();`

File: frontend/src/app/core/auth.service.spec.ts (hunk 3 above)
Comment (Toubi01) [reply frontend/src/app/core/auth.service.spec.ts:41]: added empty lines between spies and service calls

File: frontend/src/app/notes/notes.component.spec.ts (hunk 4 above)
Comment (Toubi01) [reply frontend/src/app/notes/notes.component.spec.ts]: removed `identitySpy`

File: frontend/src/app/notes/notes.component.spec.ts [hunk 5]
```diff
@@ -1,29 +1,28 @@
...
 describe('NotesComponent', () => {
-  const identitySpy = createServiceSpy('Identity');
   let component;
   let fixture;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       declarations: [NotesComponent],
-      providers: [{ provide: IdentityService, useValue: identitySpy }],
+      providers: [],
```
Comment (Marlaqk) [inline frontend/src/app/notes/notes.component.spec.ts:14]: remove providers aswell. Not needed.

File: frontend/src/app/_test-utils/spies-services.ts [hunk 6]
```diff
@@ -38,7 +38,20 @@ const spiesMethods: SpiesMethods = {
   Auth: ['runInitialLoginSequence', 'login', 'logout', 'refresh', 'hasValidToken', 'isAuthenticated$'],
   AuthGuard: ['canActivate'],
   Identity: ['getIdentity'],
```
Comment (Marlaqk) [inline frontend/src/app/_test-utils/spies-services.ts:40]: Identity can be removed from this file aswell

File: frontend/src/app/_test-utils/spies-services.ts [hunk 7]
```diff
@@ -28,7 +28,7 @@
 // Please, try to keep this in alphabetic order, so it's easier
 // to detect which service is already declared
 
-type SpiedService = 'Auth' | 'AuthGuard' | 'Identity' | 'Notes';
+type SpiedService = 'Auth' | 'AuthGuard' | 'Identity' | 'Notes' | 'Oauth';
```
Comment (Marlaqk) [inline frontend/src/app/_test-utils/spies-services.ts:31]: Identity can be removed from this file aswell

**Review by Marlaqk — APPROVED** [review]
Remove the rest of `identity`. The rest looks fine.

**Pull request merged**